### ISSUE TITLE
Fish autocompletion fails when completing tag removal

### ISSUE THREAD

**Issue by aleprovencio** [GothenburgBitFactory/taskwarrior#2619]
This happens when trying to autocomplete tags using the '-', i.e: `task <filter> modify -next`.

Depending on the tag, a different error output is shown.

**Comment by tbabej** [comment]
@aleprovencio Did this change between 2.5.3 and 2.6.0 by any chance? I remember there were some merge conflicts in the fish autocompletion that I could have messed up resolving prior to the release :thinking:

**Comment by aleprovencio** [comment]
My *guess* is that only the `next` tag used to have problems on previous versions and since 2.6.0 the bug spread to all tags.

**Comment by tbabej** [comment]
Interesting. Perhaps @soraxas or @dniku know what could be happening here?

**Comment by soraxas** [comment]
Do you want to try apply this patch to the fish completion file to see if that fixes the error?
```
$ cat task.fish.patch
```
```
--- task.fish	2021-10-09 01:33:29.884952184 +1100
+++ task.fish2	2021-10-09 01:34:22.458120141 +1100
@@ -107,7 +107,7 @@
   or return 1
   # only start completion when there's a colon in attr_name
   set -l cmd (commandline -ct)
-  string match -q "*:*" "$cmd[-1]"
+  string match -q -- "*:*" "$cmd[-1]"
 end
 
 function __fish.task.need_to_complete.command
@@ -251,13 +251,13 @@
     echo -e (string replace --all -r "^|\n" "\n$user_input_numeric" $__fish_task_static_reldates | string collect)
     # special cases for 1st, 2nd and 3rd, and 4-0th
     set -l suffix 'th' '4th, 5th, etc.'
-    if string match -q "*1" $user_input_numeric
+    if string match -q -- "*1" $user_input_numeric
       set suffix 'st' 'first'
-    else if string match -q "*2" $user_input_numeric
+    else if string match -q -- "*2" $user_input_numeric
       set suffix 'nd' 'second'
-    else if string match -q "*3" $user_input_numeric
+    else if string match -q -- "*3" $user_input_numeric
       set suffix 'rd' 'third'
-    end                  
+    end
     echo -e $user_input_numeric"$suffix[1]\t$suffix[2]"
   else
     echo -e $__fish_task_static_dates
```
apply it with
```
patch task.fish < task.fish.patch
```

**Comment by aleprovencio** [comment]
@soraxas sorry about taking so long to see this, I was off last few days.
This patch works fine, problem solved. :tada:

**Comment by tbabej** [comment]
@aleprovencio great to hear that! Let's keep this open to include this fix in 2.6.1 - @soraxas do you want to do the honors and open up a PR? :slightly_smiling_face:

**Comment by aleprovencio** [comment]
Here you go @tbabej, https://github.com/GothenburgBitFactory/taskwarrior/pull/2630
Thanks for the oportunity :+1:
